crates/util/src/paths.rs

```diff
@@ -93,6 +97,78 @@ impl<T: AsRef<Path>> PathExt for T {
 
         self.as_ref().file_name()?.to_str()?.split('.').last()
     }
+
+    /// Returns a `SanitizedPathBuf` with a cleaned-up version of the current path,
+    /// or an error if the path is empty on Windows.
+    ///
+    /// On Windows, this method removes the `\\?\` prefix from the path. On non-Windows
+    /// platforms, it directly converts the path reference into a `PathBuf`.
+    ///
+    /// Generally, this method should be used when the path is to be displayed to the user,
+    /// and the path should be returned from `canonicalize`.
+    fn sanitized_pathbuf(&self) -> anyhow::Result<SanitizedPathBuf> {
+        #[cfg(target_os = "windows")]
+        return Ok(SanitizedPathBuf(PathBuf::from(
+            self.sanitized_pathbuf_string()?,
+        )));
+        #[cfg(not(target_os = "windows"))]
+        return Ok(SanitizedPathBuf(self.as_ref().to_path_buf()));
+    }
+
+    /// Returns a sanitized string representation of the current path, or an error
+    /// if the path is empty on Windows.
+    ///
+    /// On Windows, this method removes the `\\?\` prefix. On other platforms,
+    /// it simply converts the path to a string.
+    ///
+    /// Generally, this method should be used when the path is to be displayed to the user,
+    /// and the path should be returned from `canonicalize`.
+    fn sanitized_pathbuf_string(&self) -> anyhow::Result<String> {
+        let path_string = self.as_ref().to_string_lossy();
+        #[cfg(target_os = "windows")]
+        {
+            let trimmed_path_string = path_string.trim_start_matches("\\\\?\\");
+            if trimmed_path_string.is_empty() {
+                anyhow::bail!("File path is empty!");
```

Comment:
I recall we were discussing file path length and other potential issues with such cut-offs, this is not what we're going to check and bail on here?

Reply:
My initial thought was that there might be cases like the following:

```rust
let path = PathBuf::from("\\\\?\\");
let result: SanitizedPathBuf = path.sanitized_path_buf();
```

However, I think it’s better to first check if the path is absolute, and only then perform the trim operation.